app/models/concerns/cms/addon/sns_poster.rb

```diff
@@ -0,0 +1,156 @@
+module Cms::Addon
+  module SnsPoster
+    extend ActiveSupport::Concern
+    include Cms::Content
+    extend SS::Addon
+
+   require 'twitter'
+
+    included do
+      field :fbauto,     type: String, default: 'expired'
+      field :twauto,     type: String, default: 'expired'
+      field :deleteauto, type: String, default: 'expired'
+      field :twid, type: String
+      field :twuid, type: String
+      field :fbid, type: String
+      field :fbuid, type: String
+      field :fbpid, type: String
+      permit_params :fbauto, :twauto, :deleteauto, :twid, :twuid, :fbid, :fbuid, :fbpid
+      after_generate_file { post_sns() }
+      after_remove_file { delete_sns() }
+    end
+
+    def sns_poster_fb_options
+      [
+        [I18n.t('views.options.state.active'), 'active'],
+        [I18n.t('views.options.state.expired'), 'expired'],
+      ]
+    end
+
+    def sns_poster_fb_options_ja
+      I18n.t("views.options.state.#{fbauto}")
+    end
+
+    def sns_poster_tw_options
+      [
+        [I18n.t('views.options.state.active'), 'active'],
+        [I18n.t('views.options.state.expired'), 'expired'],
+      ]
+    end
+
+    def sns_poster_tw_options_ja
+      I18n.t("views.options.state.#{twauto}")
+    end
+
+    def sns_poster_delete_options
+      [
+        [I18n.t('views.options.state.active'), 'active'],
+        [I18n.t('views.options.state.expired'), 'expired'],
+      ]
+    end
+
+    def sns_poster_delete_options_ja
+      I18n.t("views.options.state.#{deleteauto}")
+    end
+
+    def site_url
+      "#{site.url}#{filename}/"
+    end
+
+    def site_full_url
+      "#{site.full_url}#{filename}/"
+    end
+
+    def image_path
+      image_ids = file_ids.to_a.first.to_s.split("").join("/")
+      "#{site.full_url}fs/" + image_ids + "/_/#{filename}"
+    end
+
+    def tw_url
+      if twauto == "active"
+        "https://twitter.com/#{twuid}/status/#{twid}"
+      end
+    end
+
+    def fb_url
+      if fbauto == "active"
+        "https://www.facebook.com/#{fbuid}/posts/#{fbpid}"
+      end
+    end
+
+    def fbid_separator(facebook_param)
+      fbidArray = facebook_param.split("_")
+    end
+
+    def twSnskeys(snskeys)
```

Comment:
__[rubocop]__
* Use snake_case for method names. [:ref](https://github.com/bbatsov/ruby-style-guide#snake-case-symbols-methods-vars)

<!-- Comment by [SideCI](https://sideci.com) -->